Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/JobDataInteraction/DataStreamCancellationUpdater.cs

```diff
@@ -177,7 +177,7 @@ private void Debug_EnsureCanContinue(ref TInstance instance)
 
             if (m_State == UpdaterState.Ready)
             {
-                throw new InvalidOperationException($"Attempting to call {nameof(Continue)} on a {instance} but that element didn't come from this {nameof(DataStreamUpdater<TInstance>)}. Please ensure that the indexer was called first.");
+                throw new InvalidOperationException($"Attempting to call {nameof(Continue)} on a {instance} but our state is {m_State}. Most likely {nameof(Continue)} was called twice or {nameof(Resolve)} was called already for this instance. Alternatively, this instance may not belong to this {nameof(DataStreamCancellationUpdater<TInstance>)}");
```

Comment:
Better wording?

``` csharp
$"Attempting to call {nameof(Continue)} on a {instance} but the state is {m_State}. Most likely {nameof(Continue)} or {nameof(Resolve)} were called twice for this instance. Alternatively, this instance may not belong to this {nameof(DataStreamCancellationUpdater<TInstance>)}"
```

Reply:
Hmmn in this specific instance there are three reasons this could have happened:

1. You called Continue twice in a row.
2. You called Resolve once and then Continue.
3. This instance isn't part of it. 

If you called Resolve twice then you'd hit the other error message which just reverses Continue/Resolve.

Reply:
ah ok. my bad.